sycl/doc/extensions/experimental/sycl_ext_intel_runtime_buffer_location.asciidoc

```diff
@@ -0,0 +1,127 @@
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 4 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+== Status
+This is an experimental extension specification, intended to provide early
+access to features and gather community feedback.  Interfaces defined in this
+specification are implemented in {dpcpp}, but they are not finalized and may
+change incompatibly in future versions of {dpcpp} without prior notice.
+*Shipping software products should not rely on APIs defined in this
+specification.*
+
+[NOTE]
+====
+This extension is currently implemented in {dpcpp} only for FPGA devices.  Attempting to use this extension on other devices or backends may result in no effect.
+
+This serves as a temporary workaround for usm allocation to accept buffer location properties, and the formal solution will come in the future.
+====
+
+== Overview
+
+This document propose a new buffer_location runtime property that can be passed to `malloc_device`.
+
+On targets that provide more than one type of global memory, this provide users the flexibility of choosing which memory the device usm should be allocated to.
+
+This information is not a hint; it is a functional requirement of the program that must be respected.
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_RUNTIME_BUFFER_LOCATION` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+== Examples
+
+[source,c++]
+----
+array = (int *)malloc_device<int>(
+        N * sizeof(int), q,
+        property_list{sycl::ext::intel::experimental::property::usm::buffer_location(2)});
+
+sycl::queue q;
+q.parallel_for(range<1>(N), [=] (id<1> i){
+  data[i] *= 2;
+}).wait();
+----
+
+
+=== Changes to runtime properties
+
+To pass the runtime properties into malloc API, a new buffer properties is introduced.
+
+[source,c++]
+----
+namespace sycl::ext::intel::experimental::property::usm {
+
+class buffer_location {
+ public:
+  buffer_location(int location);
+  int get_buffer_location() const;
+};
+
+} // namespace sycl::ext::intel::experimental::property::usm
+----
+
```

Comment:
Add a formal description of the property here.  This description should explain what the `location` parameter means.  This is also the place to explain what happens if you specify the property for a non-FPGA device.  I was suggesting earlier that it could simply be ignored for non-FPGA devices, but I suppose we could instead say that `sycl::malloc_device()` throws an exception in this case.  I'm not sure which would be more convenient for users.  This section of the document should tell users what to expect in this scenario, though.

BTW, are there any special restrictions about how the pointer is used inside the kernel?  If there are, those need to be documented here also.